clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -204,23 +204,14 @@ public synchronized void ensureCoordinatorReady() {
                 time.sleep(retryBackoffMs);
             }
         }
+        // Since coordinator is set in a synchronized block in the group coordinator response callback,
+        // the future associated with the cooordinator request can be safely cleared here.
+        findCoordinatorFuture = null;
```

Comment:
@ijuma Setting to `null` here enables earlier garbage collection of `findCoordinatorFuture`, but in all cases including `future.exception()`, `findCoordinatorFuture.isDone()` in `lookupCoordinator` takes care of resending the request (only) when required. Setting to `null` in a finally block would enable GC sooner for the `future.exception` case, but I dont think it would remove the check for `findCoordinator.isDone()` in `lookupCoordinator` since commitAsync could have triggered a coordinator request whose state needs to be checked by the heartbeat thread. Don't mind adding a try->finally if you think it is useful.